cf/commands/servicekey/service_key_test.go

```diff
@@ -0,0 +1,125 @@
+		It("fails when not logged in", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: false}
+			Expect(callGetServiceKey([]string{"fake-service-key-name"})).To(BeFalse())
+		})
+
+		It("requires two arguments and one option to run", func() {
+			Expect(callGetServiceKey([]string{})).To(BeFalse())
+			Expect(callGetServiceKey([]string{"fake-arg-one"})).To(BeFalse())
+			Expect(callGetServiceKey([]string{"fake-arg-one", "fake-arg-two", "fake-arg-three"})).To(BeFalse())
+		})
+
+		It("fails when service instance is not found", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, ServiceInstanceNotFound: true}
+			Expect(callGetServiceKey([]string{"non-exist-service-instance"})).To(BeFalse())
+		})
+
+		It("fails when space is not targetted", func() {
+			requirementsFactory = &testreq.FakeReqFactory{LoginSuccess: true, TargetedSpaceSuccess: false}
+			Expect(callGetServiceKey([]string{"fake-service-instance", "fake-service-key-name"})).To(BeFalse())
+		})
+	})
+
+	Describe("requirements are satisfied", func() {
+		Context("gets service key successfully", func() {
+			BeforeEach(func() {
+				serviceKeyRepo.GetServiceKeyMethod.ServiceKey = models.ServiceKey{
+					Fields: models.ServiceKeyFields{
+						Name:                "fake-service-key",
+						Guid:                "fake-service-key-guid",
+						Url:                 "fake-service-key-url",
+						ServiceInstanceGuid: "fake-service-instance-guid",
+						ServiceInstanceUrl:  "fake-service-instance-url",
+					},
+					Credentials: map[string]interface{}{
+						"username": "fake-username",
+						"password": "fake-password",
+						"host":     "fake-host",
+						"port":     "3306",
+						"database": "fake-db-name",
+						"uri":      "mysql://fake-user:fake-password@fake-host:3306/fake-db-name",
+					},
+				}
+			})
+
+			It("gets service credential", func() {
+				callGetServiceKey([]string{"fake-service-instance", "fake-service-key"})
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Getting key", "fake-service-key", "for service instance", "fake-service-instance", "as", "my-user"},
+					[]string{"username", "fake-username"},
+					[]string{"password", "fake-password"},
+					[]string{"host", "fake-host"},
+					[]string{"port", "3306"},
+					[]string{"database", "fake-db-name"},
+					[]string{"uri", "mysql://fake-user:fake-password@fake-host:3306/fake-db-name"},
+					[]string{"OK"},
+				))
+				Expect(serviceKeyRepo.GetServiceKeyMethod.InstanceId).To(Equal("fake-service-instance-guid"))
+			})
+
+			It("gets service guid", func() {
+				callGetServiceKey([]string{"--guid", "fake-service-instance", "fake-service-key"})
+				Expect(ui.Outputs).To(ContainSubstrings(
+					[]string{"Getting key", "fake-service-key", "for service instance", "fake-service-instance", "as", "my-user"},
+					[]string{"fake-service-key-guid"},
+				))
+			})
+		})
+
+		It("does not show service key detail when service key not exist", func() {
```

Comment:
This one should be in different Context since the current context is: `gets service key successfully`, or remove that context.